### PR TITLE
Add format parameter to Float.parse

### LINKED ISSUE #9674: Add `format` parameter to `Float.parse` and `Decimal.parse`
The `Number.format` and `Decimal.format` methods support a `format` paramter to specify the number format for formatting. `parse` should also allow this.

### PR DESCRIPTION
**Pull request by GregoryTravis**
Closes #9674.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 1]
```diff
@@ -747,26 +747,59 @@ type Float
     to_decimal self -> Decimal = Decimal.from_float self
 
     ## ALIAS from text
-       GROUP Conversions
+       GROUP conversions
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]:
```suggestion
       GROUP Conversions
```

No idea is case sensitive but was correct as `C`

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 2]
```diff
@@ -747,26 +747,59 @@ type Float
...
-       - text: The text to parse into a float.
-       - locale: The locale that specifies the format to use when parsing
+       - text: The text to parse into a `Float`.
+       - locale: The locale that specifies the format to use when parsing.
+       - format: The Java-style format to use to parse the string.
+
+       ! Error Conditions
+
+         - If `text` is incorrectly formatted, a `Number_Parse_Error` is thrown.
+         - If `format` is incorrectly formatted (or does not match the `locale`,
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: looks like a typo

```suggestion
         - If `format` is incorrectly formatted (or does not match the `locale`),
```

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: also - what does it mean for the format to match the locale? I don't really know.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 3]
```diff
@@ -747,26 +747,59 @@ type Float
...
+           an `Illegal_Argument` is thrown.
+
+       > Example
+         Parse a `Float` with no local specifier.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: shouldn't this say locale?

```suggestion
         Parse a `Float` with no locale specifier.
```

**Review by radeusgd — APPROVED** [review]
Looks good to me

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso (hunk 3 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: Done

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: I clarified it: "(or uses digit separators that differ from the ones specified by the locale)"

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso (hunk 1 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso]: Done

**Pull request merged**